wen-restart/src/wen_restart.rs

```diff
@@ -531,91 +578,275 @@ mod tests {
+            let blockstore_clone = init_result.blockstore.clone();
+            let bank_forks_clone = init_result.bank_forks.clone();
+            let exit = Arc::new(AtomicBool::new(false));
+            let exit_clone = exit.clone();
+            let wen_restart_thread_handle = Builder::new()
+                .name("solana-wen-restart".to_string())
+                .spawn(move || {
+                    assert!(wait_for_wen_restart(
+                        &wen_restart_proto_path_clone,
+                        VoteTransaction::from(Vote::new(vec![last_vote_slot], last_vote_bankhash)),
+                        blockstore_clone,
+                        cluster_info_clone,
+                        bank_forks_clone,
+                        Some(Arc::new(RwLock::new(Vec::new()))),
+                        80,
+                        exit_clone,
+                    )
+                    .is_err());
+                })
+                .unwrap();
+            let node_pubkey = keypairs.node_keypair.pubkey();
+            let node = LegacyContactInfo::new_rand(&mut rng, Some(node_pubkey));
+            let last_vote_hash = Hash::new_unique();
+            let now = timestamp();
+            push_restart_last_voted_fork_slots(
+                init_result.cluster_info.clone(),
+                &node,
+                &expected_slots_to_repair,
+                &last_vote_hash,
+                &keypairs.node_keypair,
+                now,
+            );
+            expected_messages.insert(
+                node_pubkey.to_string(),
+                LastVotedForkSlotsRecord {
+                    last_voted_fork_slots: expected_slots_to_repair.clone(),
+                    last_vote_bankhash: last_vote_hash.to_string(),
+                    shred_version: SHRED_VERSION as u32,
+                    wallclock: now,
+                },
+            );
+            sleep(Duration::from_millis(200));
+            exit.store(true, Ordering::Relaxed);
+            let _ = wen_restart_thread_handle.join();
+            let buffer = read(&init_result.wen_restart_proto_path).unwrap();
+            let progress = WenRestartProgress::decode(&mut std::io::Cursor::new(buffer)).unwrap();
+            assert_eq!(
+                progress,
+                WenRestartProgress {
+                    state: RestartState::LastVotedForkSlots.into(),
+                    my_last_voted_fork_slots: Some(LastVotedForkSlotsRecord {
+                        last_voted_fork_slots: init_result.last_voted_fork_slots.clone(),
+                        last_vote_bankhash: last_vote_bankhash.to_string(),
+                        shred_version: SHRED_VERSION as u32,
+                        wallclock: start_time,
+                    }),
+                    last_voted_fork_slots_aggregate: Some(LastVotedForkSlotsAggregateRecord {
+                        received: expected_messages.clone(),
+                    }),
+                }
+            )
+        }
```

Comment:
This  specific `WenRestartProgress` state is all you ultimately want written to `wen_restart_proto_path` right? 

Seems like everything else, setting up the thread etc. is a complicated way to get to this end, can we just directly write the record instead?